src/lib/y2storage/proposal/assigned_space.rb

```diff
@@ -64,6 +65,43 @@ def unused
         max >= disk_size ? 0 : disk_size - max
       end
 
+      # Space available in addition to the target
+      #
+      # @return [DiskSize]
+      def extra_size
+        disk_size - volumes.target_disk_size
+      end
+
+      # Space that can be distributed among the planned volumes.
+      #
+      # Substracts from the total the space that will be used by new data
+      # structures, like the EBRs of the planned logical partitions
+      #
+      # @return [DiskSize]
+      def usable_size
+        logical = num_logical
+        # If the extended partition already exists, the overhead of our first
+        # logical partition is already counted
+        logical -= 1 if partition_type == :logical
+        disk_space.disk_size - overhead_of_logical * logical
```

Comment:
Actually it never happens because, by definition, the num_logical is always > 0 if the partition is :logical. Ahem... obviously, I need to do that more obvious in the code/comment. :smile: